cmd/oras/pull.go

```diff
@@ -103,17 +104,38 @@ func runPull(opts pullOptions) error {
 
 	// Copy Options
 	copyOptions := oras.DefaultCopyOptions
+	configFileName, configMediaType := parseFileRef(opts.ManifestConfigRef, oras.MediaTypeUnknownConfig)
+	copyOptions.FindSuccessors = func(ctx context.Context, fetcher content.Fetcher, desc ocispec.Descriptor) ([]ocispec.Descriptor, error) {
+		successors, err := content.Successors(ctx, fetcher, desc)
+		if err != nil {
+			return nil, err
+		}
+		ret := []ocispec.Descriptor{}
+		for _, s := range successors {
+
+			if s.MediaType == configMediaType {
+				// Add annotation for manifest config
+				if s.Annotations == nil {
+					s.Annotations = make(map[string]string)
+				}
+				s.Annotations[ocispec.AnnotationTitle] = configFileName
+			}
+			if s.Annotations[ocispec.AnnotationTitle] != "" {
+				// Only copy named blobs
+				ret = append(ret, s)
+			}
```

Comment:
This logic is problematic. What if the current descriptor is a manifest list / index?

Reply:
Looks like we still need a function to tell if a media can have down edge on oras-go side?

Reply:
Added a oras-project/oras-go#193 for the manifest checking on oras-go side and applied it here.

There is a corner case we cannot fix: if there is an oras reference manifest and it's `Subject` is nil, we will still fetch its content even it may be unnamed and has no successor.

Anyway, this cost cannot be avoided since the `Subject` cannot be obtained without fetching.